lib/codecs/src/decoding/framing/chunked_gelf.rs

```
@@ -0,0 +1,650 @@
+        // nor to be received at all. So, we have to store the chunks in a buffer (state field) until we receive
+        // all the chunks of a message. When we receive all the chunks of a message, we can concatenate them
+        // and return the complete payload.
+
+        // We need 10 bytes to read the message id, sequence number and total chunks
+        if chunk.remaining() < 10 {
+            let src_display = format!("{chunk:?}");
+            warn!(message = "Received malformed chunk headers (message ID, sequence number and total chunks) with less than 10 bytes. Ignoring it.",
+                src = src_display,
+                remaining = chunk.remaining(),
+                internal_log_rate_limit = true
+            );
+            return Ok(None);
+        }
+        let message_id = chunk.get_u64();
+        let sequence_number = chunk.get_u8();
+        let total_chunks = chunk.get_u8();
+
+        if total_chunks == 0 || total_chunks > GELF_MAX_TOTAL_CHUNKS {
+            warn!(
+                message = "Received a chunk with an invalid total chunks value. Ignoring it.",
+                message_id = message_id,
+                sequence_number = sequence_number,
+                total_chunks = total_chunks,
+                internal_log_rate_limit = true
+            );
+            return Ok(None);
+        }
+
+        if sequence_number >= total_chunks {
+            warn!(
+                message = "Received a chunk with a sequence number greater than total chunks. Ignoring it.",
+                message_id = message_id,
+                sequence_number = sequence_number,
+                total_chunks = total_chunks,
+                internal_log_rate_limit = true
+            );
+            return Ok(None);
+        }
+
+        let mut state_lock = self.state.lock().unwrap();
+
+        if state_lock.len() >= self.pending_messages_limit {
+            warn!(
```

About all the warnings in this method (not only this one), should we send some about them? Here we are silently discarding messages and I think it would be very useful to see how many packets are being discarded and why.

Currently, I see that the metrics custom-framework is located on `src/internal_events` and there is no easy integration with the crates under `lib/`. Other decoders does not sent any type of metric when discarding, so maybe it does not have sense here either.

---

As mentioned above, I think we should actually be returning an error in these cases. The caller of the codecs should already be handling the errors by logging and emitting metrics.